libcontainer/userns/userns_linux.go

```diff
@@ -15,22 +16,33 @@ var (
 // Originally copied from github.com/lxc/lxd/shared/util.go
 func runningInUserNS() bool {
 	nsOnce.Do(func() {
-		uidmap, err := user.CurrentProcessUIDMap()
+		file, err := os.Open("/proc/self/uid_map")
 		if err != nil {
 			// This kernel-provided file only exists if user namespaces are supported
 			return
 		}
-		inUserNS = uidMapInUserNS(uidmap)
+		defer file.Close()
+
+		buf := bufio.NewReader(file)
+		l, _, err := buf.ReadLine()
+		if err != nil {
+			return
+		}
+
+		inUserNS = uidMapInUserNS(string(l))
 	})
 	return inUserNS
 }
 
-func uidMapInUserNS(uidmap []user.IDMap) bool {
+func uidMapInUserNS(uidMap string) bool {
+	var a, b, c int64
+	_, _ = fmt.Sscanf(uidMap, "%d %d %d", &a, &b, &c)
```

Comment:
I think ignoring the error hurts readability, as whoever reads this needs to think more about what is going on here. For example, in case of an error `a` and `b` will be `0`, and we check for it. What would happen for input such as `x x 4294967295`, will `c` still be correctly assigned? (The answer is no but it's not super obvious).

I'd add a check for `err == nil` just for the sake of readability.